python/mxnet/symbol/contrib.py

```diff
@@ -137,18 +137,16 @@ def _construct_subgraph(sym_out, sym_states):
 
     flat_out = []
     all_input_names = g.list_inputs()
-    output_names = [o.name for o in sym_out]
+    output_names = {o.name for o in sym_out}
     for o in sym_out:
-        if o.name in all_input_names:
+        if o.name in all_input_names or o.list_attr().get("__subgraph_name__", "") != name:
```

Comment:
what happens if `o` doesn't have `__subgraph_name__`?

Reply:
@zheng-da `dict.get(key, defaultvalue)` will return you `defaultvalue` if `key` doesn't exist